web/packages/shared/utils/highbar.ts

```diff
@@ -0,0 +1,363 @@
+/**
+ * Copyright 2023 Gravitational, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+type SupportedMergeTypes = string | number | Record<string, unknown>;
+type MergeTarget = Record<string, unknown> | Array<SupportedMergeTypes>;
+
+export function mergeDeep(target: MergeTarget, ...sources: Array<MergeTarget>) {
+  const isObject = obj => obj && typeof obj === 'object' && !Array.isArray(obj);
+
+  const mergeArray = (target, source) => {
+    source.forEach((value, index) => {
+      if (
+        Array.isArray(value) ||
+        (isObject(value) && isObject(source[index]))
+      ) {
+        if (target[index] === undefined) {
+          target[index] = source[index];
+        } else {
+          mergeDeep(target[index], source[index]);
+        }
+      } else {
+        target[index] = source[index];
+      }
+    });
+  };
+
+  if (!sources.length) return target;
+  const source = sources.shift();
+
+  if (isObject(target) && isObject(source)) {
+    for (const key in source) {
+      if (isObject(source[key])) {
+        if (!target[key]) Object.assign(target, { [key]: {} });
+        mergeDeep(target[key], source[key]);
+      } else if (Array.isArray(source[key])) {
+        mergeArray(target[key], source[key]);
+      } else {
+        Object.assign(target, { [key]: source[key] });
+      }
+    }
+  } else if (Array.isArray(target) && Array.isArray(source)) {
+    mergeArray(target, source);
+  }
+
+  return mergeDeep(target, ...sources);
+}
+
+export function isInteger(checkVal: any): boolean {
+  return Number.isInteger(checkVal) || checkVal == parseInt(checkVal);
+}
+
+export function isObject(checkVal) {
```

Comment:
```suggestion
export function isObject(checkVal: unknown): boolean {
```

Reply:
Thanks, I'll add this change, although I was leaving it as inferred because it seemed to infer correctly. But if we prefer to use `unknown` in this case then 👍 

<img width="481" alt="Screenshot 2023-02-17 at 9 51 22 AM" src="https://user-images.githubusercontent.com/532033/219701703-dedd107b-5efa-4186-9a87-ca84de75b722.png">